### PR TITLE
[apiConformancForce] creating plugin before setting properties

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
 - *When we just call set_properties and no initialize plugin before it or indirectly(in compile_model for example), the properties will set just in PluginRegistry to structure with type of values Any and no validations will be made. So, let's add get_version to force initialization of plugin.*
 - *Please note, that commit fixes test SetPropertiesWithIncorrectKey , but after adding the function call tests SetCorrectProperties for CPU plugin will be fails, because of unsupported DeviceID, but looks like it more correct.*

### Tickets:
 - *CVS-98405*

### CODE REVIEW

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 1]
```diff
@@ -74,16 +74,19 @@ void OVSetPropComplieModleGetPropTests::SetUp() {
 }
 
 TEST_P(OVEmptyPropertiesTests, SetEmptyProperties) {
+    core->get_versions(target_device);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:77]: why do we need it here? `get_property` already creates plugin

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 2]
```diff
@@ -74,16 +74,19 @@ void OVSetPropComplieModleGetPropTests::SetUp() {
...
 }
 
 // Setting correct properties doesn't throw
 TEST_P(OVPropertiesTests, SetCorrectProperties) {
+    core->get_versions(target_device);
     OV_ASSERT_NO_THROW(core->set_property(target_device, properties));
 }
 
 TEST_P(OVPropertiesTests, canSetPropertyAndCheckGetProperty) {
+    core->get_versions(target_device);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:89]: why do we need it here? Even if we set incorrect properties, `get_property` in the loop creates plugin and can receive exceptions

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 3]
```diff
@@ -102,6 +106,7 @@ TEST_P(OVPropertiesIncorrectTests, CanNotCompileModelWithIncorrectProperties) {
 }
 
 TEST_P(OVPropertiesDefaultTests, CanSetDefaultValueBackToPlugin) {
+    core->get_versions(target_device);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:109]: `get_property` already initializes the plugin

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 4]
```diff
@@ -114,6 +119,7 @@ TEST_P(OVPropertiesDefaultTests, CanSetDefaultValueBackToPlugin) {
 }
 
 TEST_P(OVPropertiesDefaultTests, CheckDefaultValues) {
+    core->get_versions(target_device);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:122]: the same

Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp]: removed

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 3 above)
Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp]: removed

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 1 above)
Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp]: yes, thank you, I missed it about get_property

**Review by ilya-lavrenov — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Pull request merged**